### PR TITLE
feat(selinux): Remove multiple permissions from trivalent_domain, reducing attack surface

### PR DESCRIPTION
**Pull request by PhysicsIsAwesome**
Remove multiple permissions from trivalent_domain, including:

- All non-user namespace capabilities
- Some user namespace capabilities
- Unused transition permissions

### CODE REVIEW

File: files/scripts/selinux/trivalent/trivalent.te [hunk 1]
```diff
@@ -78,8 +78,7 @@ allow trivalent_domain self:sem create_sem_perms;
 allow trivalent_domain self:netlink_kobject_uevent_socket client_stream_socket_perms;
 allow trivalent_domain self:user_namespace create;
 allow trivalent_domain self:unix_stream_socket { connectto rw_socket_perms };
-allow trivalent_domain self:cap_userns { sys_admin sys_chroot sys_ptrace net_admin setpcap };
-allow trivalent_domain self:capability { dac_read_search sys_admin sys_chroot sys_ptrace };
```
Comment (RoyalOughtness) [inline files/scripts/selinux/trivalent/trivalent.te:82]: The crash reporter uses ptrace, but this should be fine for general usage

**Comment by RoyalOughtness** [comment]
@PhysicsIsAwesome can you document what tests you ran on this?

File: files/scripts/selinux/trivalent/trivalent.te (hunk 1 above)
Comment (PhysicsIsAwesome) [reply files/scripts/selinux/trivalent/trivalent.te:82]: `ptrace` is still allowed. Just the capability `sys_ptrace` outside of `userns` is removed. So I would assume crash reporter works fine, even with my changes, although I haven't tested it.

**Comment by PhysicsIsAwesome** [comment]
> @PhysicsIsAwesome can you document what tests you ran on this?

Ran Trivalent, visited a few websites, ran keepassxc extension and tested connection to the keepassxc app. Checked logs for Selinux denials. 

Although I don't see a theoretical need for the permissions I removed, some additional testing from other users might still be helpful, just to be sure to not oversee some edge case.

**Review by WavyEbuilder — COMMENTED** [review]
I can indeed speak positively on removing transition and dyntransition, that seems expected and reasonable to me and is in life with what I have policy wise.

File: files/scripts/selinux/trivalent/trivalent.te (hunk 1 above)
Comment (PhysicsIsAwesome) [reply files/scripts/selinux/trivalent/trivalent.te:82]: @RoyalOughtness I just tested crashpad with `chrome://crash`, `chrome://crashes` and `chrome://inducebrowsercrashforrealz` and it seems to work fine with my policy changes

**Review by HastD — APPROVED** [review]

**Review by RoyalOughtness — APPROVED** [review]
